src/Microsoft.DotNet.TestFramework/TestAssetInstance.cs

```diff
@@ -156,19 +174,141 @@ public TestAssetInstance UseCurrentRuntimeFrameworkVersion()
 
-                file.CopyTo(newPath);
+                CopyFileAdjustingPaths(file, newFile);
+            }
+        }
+
+        private bool IsAncestor(FileInfo file, DirectoryInfo maybeAncestor)
+        {
+            var dir = file.Directory;
+            do
+            {
+                if (string.Equals(maybeAncestor.FullName, dir.FullName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+
+                dir = dir.Parent;
+            }
+            while (dir != null);
+
+            return false;
+        }
+
+        private void CopyFileAdjustingPaths(FileInfo source, FileInfo destination)
+        {
+            if (string.Equals(source.Name, "nuget.config", StringComparison.OrdinalIgnoreCase))
+            {
+                CopyNugetConfigAdjustingPath(source, destination);
+            }
+            else
+            {
+                source.CopyTo(destination.FullName);
+            }
+        }
+
+        private void CopyNugetConfigAdjustingPath(FileInfo source, FileInfo destination)
+        {
+            var doc = XDocument.Load(source.FullName, LoadOptions.PreserveWhitespace);
+            foreach (var packageSource in doc.Root.Element("packageSources").Elements("add").Attributes("value"))
+            {
+                if (!Path.IsPathRooted(packageSource.Value))
+                {
+                    string fullPathAtSource = Path.GetFullPath(Path.Combine(source.Directory.FullName, packageSource.Value));
+                    if (!IsAncestor(new FileInfo(fullPathAtSource), TestAssetInfo.Root))
+                    {
+                        packageSource.Value = fullPathAtSource;
+                    }
+                }
+
+                using (var file = new FileStream(
+                    destination.FullName,
+                    FileMode.CreateNew,
+                    FileAccess.ReadWrite))
+                {
+                    doc.Save(file, SaveOptions.None);
+                }
+            }
+        }
+
+        private void BuildRootProjectOrSolution()
+        {
+            string[] args = new string[] { "build" };
+
+            Console.WriteLine($"TestAsset Build '{TestAssetInfo.AssetName}'");
+
+            var commandResult = Command.Create(TestAssetInfo.DotnetExeFile.FullName, args)
+                                    .WorkingDirectory(Root.FullName)
+                                    .CaptureStdOut()
+                                    .CaptureStdErr()
+                                    .Execute();
+
+            int exitCode = commandResult.ExitCode;
+
+            if (exitCode != 0)
+            {
+                Console.WriteLine(commandResult.StdOut);
+
+                Console.WriteLine(commandResult.StdErr);
+
+                string message = string.Format($"TestAsset Build '{TestAssetInfo.AssetName}' Failed with {exitCode}");
+
+                throw new Exception(message);
```

Comment:
true, I have not changed this particular piece of logic - I just moved it from one place to another - I think it would be better to do another pass on all new Exception patterns